catroid/src/org/catrobat/catroid/ui/dialogs/LegoNXTSensorPortConfigDialog.java

```diff
@@ -0,0 +1,146 @@
+import android.app.DialogFragment;
+import android.content.Context;
+import android.content.DialogInterface;
+import android.os.Bundle;
+import android.util.Log;
+
+import org.catrobat.catroid.R;
+import org.catrobat.catroid.devices.mindstorms.nxt.sensors.NXTSensor;
+import org.catrobat.catroid.ui.SettingsActivity;
+
+public class LegoNXTSensorPortConfigDialog extends DialogFragment {
+
+	public static final String DIALOG_FRAGMENT_TAG = "dialog_lego_nxt_sensor_port_config";
+	public static final String TAG = LegoNXTSensorPortConfigDialog.class.getSimpleName();
+	private int clickedItem = 0;
+	private String title;
+	private NXTSensor.Sensor nxtSensor;
+	private String[] sensorMappingStrings;
+	private String[] sensorMappings;
+	NXTSensor.Sensor[] sensorMapping;
+
+	public LegoNXTSensorPortConfigDialog(int clickedResItem) {
+		clickedItem = clickedResItem;
+	}
+
+	@Override
+	public Dialog onCreateDialog(Bundle savedInstanceState) {
+		sensorMapping = SettingsActivity.getLegoMindstormsNXTSensorMapping(this.getActivity());
+		sensorMappingStrings = getResources().getStringArray(R.array.nxt_sensor_chooser);
+		String[] portStrings = getResources().getStringArray(R.array.nxt_port_chooser);
+
+		sensorMappings = new String[4];
+
+		for (int index = 0; index < 4; index++) {
+			sensorMappings[index] = portStrings[index] + ": " + sensorMappingStrings[sensorMapping[index].ordinal()];
+		}
+
+		getAttributes(clickedItem);
+
+		Dialog dialog = new AlertDialog.Builder(getActivity())
+				.setTitle(title)
+				.setItems(sensorMappings, new DialogInterface.OnClickListener() {
+					public void onClick(DialogInterface dialog, int which) {
+						checkIfOverwrite(which);
+					}
+				}).create();
+
+		dialog.setCanceledOnTouchOutside(true);
+
+		return dialog;
+	}
+
+	private void getAttributes(int clickedItem) {
+		switch (clickedItem) {
+			case R.string.formula_editor_sensor_lego_nxt_touch:
+				title = getString(R.string.nxt_sensor_touch);
+				nxtSensor = NXTSensor.Sensor.TOUCH;
+				break;
+			case R.string.formula_editor_sensor_lego_nxt_sound:
+				title = getString(R.string.nxt_sensor_sound);
+				nxtSensor = NXTSensor.Sensor.SOUND;
+				break;
+			case R.string.formula_editor_sensor_lego_nxt_light:
+				title = getString(R.string.nxt_sensor_light);
+				nxtSensor = NXTSensor.Sensor.LIGHT_INACTIVE;
+				break;
+			case R.string.formula_editor_sensor_lego_nxt_light_active:
+				title = getString(R.string.nxt_sensor_light_active);
+				nxtSensor = NXTSensor.Sensor.LIGHT_ACTIVE;
+				break;
+			case R.string.formula_editor_sensor_lego_nxt_ultrasonic:
+				title = getString(R.string.nxt_sensor_ultrasonic);
+				nxtSensor = NXTSensor.Sensor.ULTRASONIC;
+				break;
+			default:
+				title = getString(R.string.nxt_sensor_not_found);
+				nxtSensor = NXTSensor.Sensor.NO_SENSOR;
+		}
+	}
+
+	private void checkIfOverwrite(final int selectedPort) {
+		String selected = sensorMappingStrings[sensorMapping[selectedPort].ordinal()];
+		//delete Context import
```

Comment:
could you use a method named `deleteContextImport(...)` instead of this comment?